bip-schnorr.mediawiki

```diff
@@ -124,7 +124,7 @@ Input:
 * The signatures ''sig<sub>1..u</sub>'': ''u'' 64-byte arrays
 
 All provided signatures are valid with overwhelming probability if and only if the algorithm below does not fail.
-* Generate ''u-1'' random integers ''a<sub>2...u</sub>'' in the range ''1...n-1''. These can either be generated deterministically using a [https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator CSPRNG] seeded by a cryptographic hash (e.g., SHA256) of all inputs of the algorithm, or be randomly generated independently for each run of the batch verification algorithm.
+* Generate ''u-1'' random integers ''a<sub>2...u</sub>'' in the range ''1...n-1''. They are generated deterministically using a [https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator CSPRNG] seeded by a cryptographic hash of all inputs of the algorithm, i.e. ''seed = hash(pk<sub>1</sub>..pk<sub>u</sub> || m<sub>1</sub>..m<sub>u</sub> || sig<sub>1</sub>..sig<sub>u</sub> )''<ref>For example, in the [https://github.com/bitcoin-core/secp256k1/pull/558 bip-schnorr implementation of the secp256k1 C library] the seed is computed from the inputs with SHA256 and fed into the [https://tools.ietf.org/html/rfc8439 ChaCha20] algorithm to generate a key stream of 256 bit random integers.</ref>.
```

Comment:
* "They are generated" -> "They should be generated"
 * Can we just say "SHA256" hash instead of cryptographic hash? It's simpler. 
 * I think inlinng is the ChaCha thing is better. Maybe "A safe choice is ChaCha20 with an IV... bla". We can still note that libsecp256k1 does that in the footnote if you don't want to say that in the body.